### PR TITLE
609 alert on reseting edit with changes

### LINKED ISSUE #609: Forms: clicking edit button throws alert
Alert could be something like: "Reitin suunnalla on tallentamattomia muutoksia. Oletko varma, että haluat lopettaa muokkaamisen? Tallentamattomat muutokset kumotaan."

### PR DESCRIPTION
**Pull request by kronken**
Closes #609

To test:
Modify and form, and click to stop editing.

Also:
- Make RoutePathHeader in to a function component
- Write number inputs as number, not as string

### CODE REVIEW

File: src/components/sidebar/ViewHeader.tsx [hunk 1]
```diff
@@ -10,19 +10,31 @@ interface IViewHeaderProps {
     hideCloseButton?: boolean;
     isEditButtonVisible?: boolean;
     isEditing?: boolean;
+    preventFromReseting?: boolean;
```
Comment (Divinare) [inline src/components/sidebar/ViewHeader.tsx]: I would say shouldShowClosePrompMessage

File: src/components/sidebar/ViewHeader.tsx [hunk 2]
```diff
@@ -10,19 +10,31 @@ interface IViewHeaderProps {
...
     onEditButtonClick?: () => void;
     onCloseButtonClick?: () => void;
 }
 
-const viewHeader = (props:IViewHeaderProps) => {
+const ViewHeader = (props:IViewHeaderProps) => {
+    // tslint:disable:max-line-length
+    const closePromptMessage = 'Tietueella on tallentamattomia muutoksia. Oletko varma, että haluat poistua näkymästä? Tallentamattomat muutokset kumotaan.';
```
Comment (Divinare) [inline src/components/sidebar/ViewHeader.tsx]: Tietueella -> Sinulla

**Comment by Divinare** [comment]
Did you test this carefully?

Make required field as empty, see save button status (should be grey)

**Comment by kronken** [comment]
Those bugs should now be fixed.

**Pull request merged**